sycl/source/detail/image_accessor_util.cpp

```diff
@@ -54,6 +109,43 @@ cl_int4 getPixelCoordNearestFiltMode(cl_float4 Coord_uvw,
   return cl_int4{i, j, k, 0};
 }
 
+bool isOutOfRange(cl_int4 PixelCoord, addressing_mode SmplAddrMode,
+                  range<3> ImgRange) {
+
+  bool OutOfRange = false;
+  cl_int u = PixelCoord.x();
+  cl_int v = PixelCoord.y();
+  cl_int w = PixelCoord.z();
+
+  cl_int width = ImgRange[0];
+  cl_int height = ImgRange[1];
+  cl_int depth = ImgRange[2];
+
+  if (SmplAddrMode == addressing_mode::clamp &&
```

Comment:
Let's write shortly(e.g.):
```
if (SmplAddrMode != addressing_mode::clamp)
    return false;
    
auto IsOutOfRange = [](cl_int V, cl_int Range) { return (V < 0) || (V >= Range); };

bool WidthOutOfRange = IsOutOfRange(PixelCoord.x(), ImgRange[0]);
bool HeightOutOfRange = IsOutOfRange(PixelCoord.y(), ImgRange[1]);
bool DepthOutOfRange = IsOutOfRange(PixelCoord.z(), ImgRange[2]);

return WidthOutOfRange ||  HeightOutOfRange || DepthOutOfRange;
```

Reply:
Very Nice.. Thanks for teaching some new C++ with every review..